src/test/unit/model/indexing/lvalue_varmat_test.cpp

```diff
@@ -0,0 +1,760 @@
+  check_vector_adjs(check_i, x, "lhs");
+  EXPECT_MATRIX_EQ(y.adj(), Vec::Ones(2));
+}
+
+TEST_F(VarAssign, minmax_vec) { test_minmax_vec<Eigen::VectorXd>(); }
+
+TEST_F(VarAssign, minmax_rowvec) { test_minmax_vec<Eigen::RowVectorXd>(); }
+
+template <typename Vec>
+void test_max_vec() {
+  using stan::math::sum;
+  using stan::math::var_value;
+  using stan::model::test::check_vector_adjs;
+  using stan::model::test::generate_linear_var_vector;
+  auto x = generate_linear_var_vector<Vec>(5);
+  Vec x_val = x.val();
+  auto y = generate_linear_var_vector<Vec>(2, 10);
+
+  assign(x, index_list(index_max(2)), y);
+  EXPECT_FLOAT_EQ(y.val()[0], x.val()[0]);
+  EXPECT_FLOAT_EQ(y.val()[1], x.val()[1]);
+  stan::math::sum(x).grad();
+  EXPECT_MATRIX_EQ(x.val(), x_val);
+  auto check_i = [](int i) { return i > 1; };
+  check_vector_adjs(check_i, x, "lhs");
+  EXPECT_MATRIX_EQ(y.adj(), Vec::Ones(2));
+}
+
+TEST_F(VarAssign, max_vec) { test_max_vec<Eigen::VectorXd>(); }
+
+TEST_F(VarAssign, max_rowvec) { test_max_vec<Eigen::RowVectorXd>(); }
+
+template <typename Vec>
+void test_min_vec() {
+  using stan::math::sum;
+  using stan::math::var_value;
+  using stan::model::test::check_vector_adjs;
+  using stan::model::test::generate_linear_var_vector;
+  auto x = generate_linear_var_vector<Vec>(5);
+  auto y = generate_linear_var_vector<Vec>(3, 10);
+  Vec x_val(x.val());
+  assign(x, index_list(index_min(3)), y);
+  EXPECT_FLOAT_EQ(x.val()(2), y.val()(0));
+  EXPECT_FLOAT_EQ(x.val()(3), y.val()(1));
+  EXPECT_FLOAT_EQ(x.val()(4), y.val()(2));
+  sum(x).grad();
+  EXPECT_MATRIX_EQ(x.val(), x_val);
+  auto check_i = [](int i) { return i < 2; };
+  check_vector_adjs(check_i, x, "lhs");
+  EXPECT_MATRIX_EQ(y.adj(), Vec::Ones(3));
+  test_throw_out_of_range(x, index_list(index_min(0)), y);
+}
+TEST_F(VarAssign, min_vec) { test_min_vec<Eigen::VectorXd>(); }
+
+TEST_F(VarAssign, min_rowvec) { test_min_vec<Eigen::RowVectorXd>(); }
+
+template <typename Vec>
+void test_omni_vec() {
+  using stan::math::sum;
+  using stan::math::var_value;
+  using stan::model::test::check_vector_adjs;
+  using stan::model::test::generate_linear_var_vector;
+  auto x = generate_linear_var_vector<Vec>(5);
+  auto y = generate_linear_var_vector<Vec>(5, 10);
+  Vec y_val = y.val();
+
+  assign(x, index_list(index_omni()), y);
+  EXPECT_FLOAT_EQ(y.val()[0], x.val()[0]);
+  EXPECT_FLOAT_EQ(y.val()[1], x.val()[1]);
+  EXPECT_FLOAT_EQ(y.val()[2], x.val()[2]);
+  EXPECT_FLOAT_EQ(y.val()[3], x.val()[3]);
+  EXPECT_FLOAT_EQ(y.val()[4], x.val()[4]);
+  sum(x).grad();
+  EXPECT_MATRIX_EQ(x.val(), y_val);
+  auto check_i = [](int i) { return true; };
+  check_vector_adjs(check_i, x, "lhs");
+  EXPECT_MATRIX_EQ(y.adj(), Vec::Ones(5));
+}
+
+TEST_F(VarAssign, omni_vec) { test_omni_vec<Eigen::VectorXd>(); }
+
+TEST_F(VarAssign, omni_rowvec) { test_omni_vec<Eigen::RowVectorXd>(); }
+
+template <typename Vec>
+void test_positive_minmax_varvector() {
```

Comment:
In what way is this test different than `test_minmax_vec`?

Reply:
Good catch I'll delete `test_minmax_vec`